website/integrations: pgadmin: refactor

dominic-r opened the pull request:
sdko/integrations/pgadmin/refactor

@tanberry yeehaw 🤠
<!--
👋 Hi there! Welcome.

Please check the Contributing guidelines: https://docs.goauthentik.io/docs/developer-docs/#how-can-i-contribute
-->

## Details

<!--
Explain what this PR changes, what the rationale behind the change is, if any new requirements are introduced or any breaking changes caused by this PR.

Ideally also link an Issue for context that this PR will close using `closes #`
-->
REPLACE ME

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

Commit:
website/integrations: pgadmin: refactor

Refactors the pgAdmin integration documentation, makes it match existing style guide, and adds subsection for configuration for containerized deployments

Commit:
website/integrations: pgadmin: lint

Lints refactored documentation page

Commit:
Merge branch 'goauthentik:main' into sdko/integrations/pgadmin/refactor

tanberry commented on website/integrations/services/pgadmin/index.md:
```diff
@@ -13,89 +13,82 @@ sidebar_label: pgAdmin
-:::note
-This is based on authentik 2022.3.3 and pgAdmin4 6.19
-:::
-
```
Assuming you updated the docs to reflect a more recent version of pgAdmin? If so, we should perhaps state that here.

tanberry replied:
Good question... yeah true most other integration guides don't have it. But, it is useful data, so let's not leave it out. Anythng useful to our dear readers, let's include. Indeed, maybe one day we could get it added to the others. ;-)

tanberry commented on website/integrations/services/pgadmin/index.md:
```diff
@@ -13,89 +13,82 @@ sidebar_label: pgAdmin
+1. From the Admin interface, navigate to **Applications** -> **Applications**.
+2. Use the wizard to create a new application and provider. During this process:
+    - Note the **Client ID**, **Client Secret**, and **slug** values because they will be required later.
+    - Set the redirect URI to `https://pgadmin.company`.
```
We now have a fancy new **Redirect URI** field... one can chose `strict` or `regex`... might be good to specify this is a strict value, if so?

tanberry commented on website/integrations/services/pgadmin/index.md:
```diff
@@ -13,89 +13,82 @@ sidebar_label: pgAdmin
+
+## pgAdmin OAuth Configuration
+
+To configure OAuth in pgAdmin, you can either use the `config_local.py` file or set environment variables if you are deploying pgAdmin in a containerized setup.
```
Do we ned to tell them exactly where to add the environment variables, as in which file? If we can, definitely let's do. Can be quite frustrating for users to be told to do something, but not where to do it.

tanberry replied:
oh, I see you have it right below... ok.

tanberry commented on website/integrations/services/pgadmin/index.md:
```diff
@@ -13,89 +13,82 @@ sidebar_label: pgAdmin
+    - If the file does not exist, create it manually.
 
-In the code above the following placeholders have been used:
+2. Add the following configuration to `config_local.py`:
```
```suggestion
2. Add the following configuration settings to `config_local.py`:
```

tanberry commented on website/integrations/services/pgadmin/index.md:
````diff
@@ -13,89 +13,82 @@ sidebar_label: pgAdmin
+    ```python
+    OAUTH2_AUTO_CREATE_USER = False
+    ```
+    Setting this value to `False` disables automatic user creation. This ensures that only the first signed-in user is registered.
````
Can we get a final section for verification? :-)

Commit:
Update website/integrations/services/pgadmin/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <git@sdko.org>

Commit:
website/integrations: pgadmin: updates note

Adds back "based on" note with updated versions

Commit:
website/integrations: pgadmin: specify strict/regex rediect uri

Not sure on wording tho

Commit:
website/integrations: pgadmin: add configuration validation step

Adds configuration validation step. Log out, log back in with authentik. Button on login page

Commit:
Merge branch 'goauthentik:main' into sdko/integrations/pgadmin/refactor

tanberry reviewed (approved):
Thank you so much! I rather do like your short-and-sweet instructions for the Wizard. (This integration is a lot less complex than other one, with 4 pairs.)

Commit:
website/integrations: pgadmin: fix redirect uri

Fixes incorrect redirect uri introduced during refactor. Probably forgot to copy slug or something. Important thing is that it's fixed

Signed-off-by: 4d62 <git@sdko.org>

Commit:
website/integrations: pgadmin: fix another stupidity I probably made

Glorious https://img.sdko.org/u/0k3f46.png

Signed-off-by: 4d62 <git@sdko.org>

Commit:
Merge branch 'goauthentik:main' into sdko/integrations/pgadmin/refactor

Commit:
Merge branch 'goauthentik:main' into sdko/integrations/pgadmin/refactor

tanberry reviewed (approved):
Thanks @4d62 !
